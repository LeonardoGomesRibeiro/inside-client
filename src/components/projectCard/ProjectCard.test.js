import React from 'react';
import { Card, Meta } from 'antd';
import { shallow } from 'enzyme';
import Project from './ProjectCard';

describe('Project Card', () => {
  const wrapper = shallow(<Project />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('renders card component properly', () => {
    expect(wrapper.find(Card).length).toBe(1);
  });
  it('renders card actions properly', () => {
    expect(wrapper.getElement().props.actions.length).toBe(3);
  });
  it('renders card component children properly', () => {
    const cardWrapper = wrapper.find(Card);
    expect(cardWrapper.children(Meta).length).toBe(1);
  });
});
