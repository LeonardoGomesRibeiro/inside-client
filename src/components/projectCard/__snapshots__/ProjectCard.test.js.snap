// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Card renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProjectCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "actions": Array [
        <Icon
          type="setting"
/>,
        <Icon
          type="edit"
/>,
        <Icon
          type="ellipsis"
/>,
      ],
      "children": <Unknown
        avatar={
                <Avatar
                        prefixCls="ant-avatar"
                        shape="circle"
                        size="default"
                        src="https://gw.alipayobjects.com/zos/rmsportal/JiqGstEfoWAOHiTxclqi.png"
                />
        }
        description="This is the description"
        title="Card title"
/>,
      "cover": <img
        alt="example"
        src="https://gw.alipayobjects.com/zos/rmsportal/JiqGstEfoWAOHiTxclqi.png"
/>,
      "style": Object {
        "width": 300,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "avatar": <Avatar
          prefixCls="ant-avatar"
          shape="circle"
          size="default"
          src="https://gw.alipayobjects.com/zos/rmsportal/JiqGstEfoWAOHiTxclqi.png"
/>,
        "description": "This is the description",
        "title": "Card title",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "actions": Array [
          <Icon
            type="setting"
/>,
          <Icon
            type="edit"
/>,
          <Icon
            type="ellipsis"
/>,
        ],
        "children": <Unknown
          avatar={
                    <Avatar
                              prefixCls="ant-avatar"
                              shape="circle"
                              size="default"
                              src="https://gw.alipayobjects.com/zos/rmsportal/JiqGstEfoWAOHiTxclqi.png"
                    />
          }
          description="This is the description"
          title="Card title"
/>,
        "cover": <img
          alt="example"
          src="https://gw.alipayobjects.com/zos/rmsportal/JiqGstEfoWAOHiTxclqi.png"
/>,
        "style": Object {
          "width": 300,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "avatar": <Avatar
            prefixCls="ant-avatar"
            shape="circle"
            size="default"
            src="https://gw.alipayobjects.com/zos/rmsportal/JiqGstEfoWAOHiTxclqi.png"
/>,
          "description": "This is the description",
          "title": "Card title",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
